Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/CoronaTestProcessor.kt

```diff
@@ -98,15 +98,34 @@ class CoronaTestProcessor @Inject constructor(
     }
 
     private fun createServerRequest(qrCode: CoronaTestQRCode): RegistrationRequest {
-        val dateOfBirthKey = if (qrCode.isDccConsentGiven && qrCode.dateOfBirth != null) {
-            DateOfBirthKey(qrCode.registrationIdentifier, qrCode.dateOfBirth!!)
-        } else null
-
-        return RegistrationRequest(
-            key = qrCode.registrationIdentifier,
-            dateOfBirthKey = dateOfBirthKey,
-            type = VerificationKeyType.GUID,
-        )
+        return when (qrCode) {
+            is CoronaTestQRCode.PCR -> {
+                val dateOfBirth = qrCode.dateOfBirth
+                val dateOfBirthKey = if (qrCode.isDccConsentGiven && dateOfBirth != null) {
+                    DateOfBirthKey(qrCode.qrCodeGUID, dateOfBirth)
+                } else null
+
+                RegistrationRequest(
+                    key = qrCode.qrCodeGUID,
+                    dateOfBirthKey = dateOfBirthKey,
```

Comment:
override val registrationIdentifier: String
            get() = qrCodeGUID

This should have worked for all types, or did it not?

Reply:
it did not